iceoryx_posh/test/moduletests/test_roudi_portmanager_client_server.cpp

```diff
@@ -0,0 +1,672 @@
+
+    EXPECT_THAT(clientPortUser1.getConnectionState(), Eq(ConnectionState::CONNECTED));
+    EXPECT_THAT(clientPortUser2.getConnectionState(), Eq(ConnectionState::CONNECTED));
+}
+
+TEST_F(PortManager_test, ServerWithBlockProducerAndClientWithWaitForConsumerAreConnected)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8c3b7770-13e6-4003-aa9f-b04a34df67c9");
+    auto clientOptions = createTestClientOptions();
+    clientOptions.serverTooSlowPolicy = ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
+    auto serverOptions = createTestServerOptions();
+    serverOptions.requestQueueFullPolicy = QueueFullPolicy::BLOCK_PRODUCER;
+
+    auto clientPortUser1 = createClient(clientOptions);
+    auto serverPortUser = createServer(serverOptions);
+    auto clientPortUser2 = createClient(clientOptions);
+
+    EXPECT_THAT(clientPortUser1.getConnectionState(), Eq(ConnectionState::CONNECTED));
+    EXPECT_THAT(clientPortUser2.getConnectionState(), Eq(ConnectionState::CONNECTED));
+}
+
+TEST_F(PortManager_test, ServerWithBlockProducerAndClientWithDiscardOldestDataAreNotConnected)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "1d89fa87-3628-4645-9147-82f4223e878a");
+    auto clientOptions = createTestClientOptions();
+    clientOptions.serverTooSlowPolicy = ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA;
+    auto serverOptions = createTestServerOptions();
+    serverOptions.requestQueueFullPolicy = QueueFullPolicy::BLOCK_PRODUCER;
+
+    auto clientPortUser1 = createClient(clientOptions);
+    auto serverPortUser = createServer(serverOptions);
+    auto clientPortUser2 = createClient(clientOptions);
+
+    EXPECT_THAT(clientPortUser1.getConnectionState(), Ne(ConnectionState::CONNECTED));
+    EXPECT_THAT(clientPortUser2.getConnectionState(), Ne(ConnectionState::CONNECTED));
+}
+
+// END policy based connection tests
+
+// BEGIN communication tests
+
+TEST_F(PortManager_test, ConnectedClientCanCommunicateWithServer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6376b58d-a796-4cc4-9c40-0c5a117b53f5");
+    auto clientOptions = createTestClientOptions();
+    clientOptions.connectOnCreate = true;
+    auto serverOptions = createTestServerOptions();
+    serverOptions.offerOnCreate = true;
+
+    auto serverPortUser = createServer(serverOptions);
+    auto clientPortUser = createClient(clientOptions);
+
+    using DataType = uint64_t;
+    constexpr int64_t SEQUENCE_ID{42};
+
+    auto allocateRequestResult = clientPortUser.allocateRequest(sizeof(DataType), alignof(DataType));
+    ASSERT_FALSE(allocateRequestResult.has_error());
+    auto requestHeader = allocateRequestResult.value();
+    requestHeader->setSequenceId(SEQUENCE_ID);
+    clientPortUser.sendRequest(requestHeader);
+
+    auto getRequestResult = serverPortUser.getRequest();
+    ASSERT_FALSE(getRequestResult.has_error());
+    auto receivedRequestHeader = getRequestResult.value();
+    EXPECT_THAT(receivedRequestHeader->getSequenceId(), Eq(SEQUENCE_ID));
+
+    auto allocateResponseResult =
```

Comment:
Do we require tests for the following case:
* serverPortUser calls allocateResponse multiple times for the same receivedRequestHeader
* `sendRequest`, `sendResponse` is called multiple times for the same header
* client sends two requests A, B and server sends responses in opposite order B, A.
* client sends so many requests (A,B,C,D) that the clients response queue (of size 2) overflows as expected so that only (C, D) responses are present.
* server is blocked when client response queue is full and the policy states blocking
* server is not blocked when client response queue is full and the policy states not blocking.

Or are those tests already implemented somewhere else and I overlooked them?